app/controllers/allocations_controller.rb

```diff
@@ -25,4 +25,15 @@ def create
 
     head 201
   end
+
+  def download
+    if current_user.role.name == "Admin"
```

Comment:
Instead of using a condition like so here, you could create a `before_action` that runs a function with this condition for a specified set of methods (just download in this case).
You can see an example in the `UsersController` with `authenticate_admin!`

Reply:
Sure I will add this!